internal/xds/translator/translator.go

```diff
@@ -613,17 +618,42 @@ func buildXdsUpstreamTLSCASecret(tlsConfig *ir.TLSUpstreamConfig) *tlsv3.Secret
 
 func buildXdsUpstreamTLSSocketWthCert(tlsConfig *ir.TLSUpstreamConfig) (*corev3.TransportSocket, error) {
 
-	tlsCtx := &tlsv3.UpstreamTlsContext{
-		CommonTlsContext: &tlsv3.CommonTlsContext{
-			TlsCertificateSdsSecretConfigs: nil,
-			ValidationContextType: &tlsv3.CommonTlsContext_ValidationContextSdsSecretConfig{
-				ValidationContextSdsSecretConfig: &tlsv3.SdsSecretConfig{
-					Name:      tlsConfig.CACertificate.Name,
-					SdsConfig: makeConfigSource(),
+	var tlsCtx *tlsv3.UpstreamTlsContext
+
+	if tlsConfig.UseSystemTrustStore {
+		tlsCtx = &tlsv3.UpstreamTlsContext{
+			CommonTlsContext: &tlsv3.CommonTlsContext{
+				ValidationContextType: &tlsv3.CommonTlsContext_ValidationContext{
+					ValidationContext: &tlsv3.CertificateValidationContext{
+						TrustedCa: &corev3.DataSource{
+							Specifier: &corev3.DataSource_Filename{
+								// This is the default location for the system trust store
+								// on Debian derivatives like the envoy-proxy image being used by the infrastructure
```

Comment:
Is that also the correct path for distroless? I think that all EG releases are now based on distroless.

Reply:
Yes, `distroless` is a Debian derivative.  See [here](https://github.com/GoogleContainerTools/distroless?tab=readme-ov-file#how-do-i-use-distroless-images).